src/components/Form/Form.tsx

```diff
@@ -344,14 +344,16 @@ const TextInputsGroup = ({
     }
   }
 
-  const groupId = uniqueId("text-inputs-group");
-
   return (
     <>
       {rows.map((rowFields, r) => {
+        // TODO: row should have a stable field to use as the key, since the key
+        // will be incorrect if the rows are shuffled. I've used the index for
+        // now since it's more (but not totally) correct than the previous
+        // behavior of using a generated id that changed on each render.
```

Comment:
I looked at `rowFields` and there didn't appear to be such a key. Hence the TODO.